Additional workflows

zotoMIT opened the pull request:
#### Developer Checklist

- [X] The README contains any additional info required
- [NA] Any special variables have values configured in AWS SSM
- [NA] Stakeholder approval has been confirmed (or is not needed)

#### What does this PR do?

This PR adds additional workflows for:
* checkov checks (terraform)
* docs generation (terraform)

And updates the readme and template.  

#### Helpful background context

These two workflows add some additional structure around terraform, giving us some automatic security checks and automatic documentation added to the readme.  
They shouldn't be active for ansible code, but if the repo is solely ansible, we may choose to just delete the workflows after the repo is created, including the tfvalidate one, and replace them with ansible specific checks.  

As with all updates to our template, this will require either deleting and recreating all our repos created by the template, or updating each repo by hand.  Since these are workflow changes that do not effect the terraform cloud workspaces, i think a manual copy would be easiest in this case.  

Additional info about checkov can be found here: https://www.checkov.io/4.Integrations/GitHub%20Actions.html
Additional info about docs generation can be found here: https://terraform-docs.io/how-to/github-action/

As a side effect of this change, because the docs generation automatically injects the terraform docs markdown, we no longer need to run that command by hand during readme updating, and should not include it there ourselves.  


#### What are the relevant tickets?

NA

#### Requires Database Migrations?

NO

#### Includes new or updated dependencies?

NO

zotoMIT committed:
Add additional workflows and update readme

# Subject
Add additional workflows for:
checkov checks (terraform)
Docs generation (terraform)

# Why these changes are being introduced:
These two workflows add some additional structure around terraform, giving us some automatic security checks and automatic documentation added to the readme.

# How this addresses that need:
I add the two workflows, and update the readme so that when the workflow is active, it will not duplicate info

# Side effects of this change:
We will get terraform docs injected to the readme in any repo that has this workflow active.  I think we just manually remove the workflow if the repo contains only ansible.

# Relevant ticket(s):
NA

zotoMIT committed:
Update versions.tf

touch a tf file to kick off actions

github-actions[bot] committed:
terraform-docs: automated action

cabutlermit commented on .github/workflows/checkov.yaml:
```diff
@@ -0,0 +1,29 @@
+## This github action runs Checkov against any terraform in the base directory. 
```
I think that this action needs a little more documentation at the top that includes a link to the Checkov site and a brief summary of what this security check actually looks for.

zotoMIT replied:
Added some more info here

cabutlermit commented on .github/workflows/checkov.yaml:
```diff
@@ -0,0 +1,29 @@
+          directory: .
+          framework: terraform 
+          quiet: true
+          #skip_check: CKV_AWS_01
```
my usual pet peeve around blank lines at ends of files 😁

zotoMIT replied:
fixed

cabutlermit commented on README.md:
````diff
@@ -27,13 +27,28 @@ After deploying this, the following steps must be completed.
 ```
 
+
+
````
Maybe a brief in-line comment that makes clearer that this section will be built automatically by a GitHub Action and should be touched otherwise?

zotoMIT replied:
Added clarification here.

cabutlermit commented on README.md:
```diff
@@ -45,4 +60,9 @@ After deploying this, the following steps must be completed.
+## Outputs
+
+No outputs.
+<!-- END_TF_DOCS -->
```
Do we want folks to be able to add more notes to the README after the auto-injected text from `terraform-docs`? Might be worth a comment here making that clear.

zotoMIT replied:
Added that up before this section.

cabutlermit reviewed (changes requested):
Overall, this is good, but there are a few places where I feel like there should be a little more inline documentation.

zotoMIT committed:
Fix spacing, and add documentation for checkov and readme

Add a space to end of files which did not have it.
Add more info about checkov, and what it checks.
Add Clarification on how the readme should be used.

cabutlermit reviewed (approved):
Looks great!  I love the Checkov integration and the automated terraform-docs insertion!
